### ISSUE TITLE
dynamic v-for to populate options

### ISSUE THREAD

**Issue by stephenhowells** [rhyek/vue2-selectize#2]
I've tried to figure out how I can get Selectize to work when creating the options list with `v-for`:

```vue
<option v-for="person in people" :value="person.id">{{ person.name }}</option>
```

The Selectize dropdown initializes as expected on the page but it's empty. Selectize seems to disregard what Vue is doing on the page. Perhaps Vue is creating the options after Selectize initializes? Not sure if that's the case or how I'd fix that. Anyway, is there a way to get `v-for` and Selectize to play nicely? Thanks.

**Comment by stephenhowells** [comment]
It seems the best way to handle this is to pass in the people array as a prop and programmatically add the options using Selectize's API. `v-for` doesn't make any sense in this regard. After thinking it through, it's pretty obvious Selectize intends for dynamic option creation to occur through their API as opposed to the plugin constantly monitoring the DOM. Would you be interested in a pull request for this, Carlos? It's sort of an odd configuration as it requires the Selectize element to be created in the watcher for the prop once the prop array populates.

**Comment by rhyek** [comment]
Hi, @stephenhowells. IMO the component should also handle updates to the `option` DOM elements from a `v-for` to make it more intuitive/practical. It should probably be done using the `render` function, but I haven't really looked into it.

Options should also update if you passed an `options` array that is later modified.

Recreating/reinitializing the `DOM` element/selectize doesn't sound like the right approach. `Selectize` has API methods such as `addOption` and `refreshOptions` which seem appropriate to me.

I would gladly accept a pull request that worked this way.

**Comment by stephenhowells** [comment]
@rhyek I didn't consider that the options prop could be modified. That makes total sense to add and refresh in the prop watcher instead of a reinitialization. I'll tinker with it and submit a PR for a dynamic options prop. I also agree that the `v-for` seems obvious and intuitive. It threw me off at first when Selectize didn't observe the options loop. I can't deny it would be convenient for `v-for` to work. If it appears straightforward with `render` or something else I'll include it in the PR. Thanks for this component!

**Comment by rhyek** [comment]
I just fixed this in 1.0.3. Updates only for `v-for` changes and not for an options array given through `settings.option`. Let me know if this works for you, please. If not I'll reopen.

**Comment by stephenhowells** [comment]
Sorry for the delay. Updates on `v-for` sounds great! That works for me.
